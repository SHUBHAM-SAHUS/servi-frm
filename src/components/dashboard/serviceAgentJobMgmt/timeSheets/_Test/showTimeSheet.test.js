import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import ShowTimeSheet from '../showTimeSheet';

describe('ShowTimeSheet Component Test', () => {
    const initialState = {
        timeSheet: {
            jobsList: [],
            timeSheetList: [],
            invoices: []
        },
        sAIncidentManagement: {
            allIncidentReportsByJob: [],
            allHazardReportsByJob: [],
            incidentReportDetails: [],
            hazardReportDetails: [],
            incidentCategories: [],
            riskControls: [],
            hazardDetails: []
        },
        sAJobMgmt: {
            jobDetails: [],
            jobReports: [],
            filePath: ''
        },
        organizationUsers: {
            usersList: []
        },
        invoice: {
            invoiceDetails: []
        }, form: {
            showTimeSheet: {}
        }
    }

    const mockStore = configureStore()
    let store, container

    beforeEach(() => {
        store = mockStore(initialState)
        container = shallow(<ShowTimeSheet store={store} />)
    })

    it('should render component ShowTimeSheet correctly', () => {
        const component = shallow(<ShowTimeSheet store={store} />);
        expect(component).toMatchSnapshot();
    });

    it('Render the connected(SMART) ShowTimeSheet component', () => {
        expect(container.length).toEqual(1)
    });
})