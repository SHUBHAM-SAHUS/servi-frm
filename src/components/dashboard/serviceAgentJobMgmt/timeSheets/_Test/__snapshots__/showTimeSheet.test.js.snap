// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowTimeSheet Component Test should render component ShowTimeSheet correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UNDEFINED
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <ForwardRef
        SAIncidentReportAction={
          Object {
            "AssignCorrectiveAction": [Function],
            "AssignHazard": [Function],
            "addHazardReport": [Function],
            "addIncidentReport": [Function],
            "getActionAssign": [Function],
            "getAllBeforeConsequences": [Function],
            "getAllHazards": [Function],
            "getAllHazardsByJobId": [Function],
            "getAllHazardsNew": [Function],
            "getAllIncidents": [Function],
            "getAllIncidentsByJobId": [Function],
            "getAllLikelihoodBeforeControl": [Function],
            "getHazardDetails": [Function],
            "getHazardDetailsNew": [Function],
            "getIncidentCategories": [Function],
            "getIncidentDetails": [Function],
            "getRiskControls": [Function],
          }
        }
        allHazardReportsByJob={Array []}
        allIncidentReportsByJob={Array []}
        destroyOnUnmount={true}
        enableReinitialize={true}
        filePath=""
        forceUnregisterOnUnmount={false}
        form="showTimeSheet"
        formValues={Object {}}
        getFormState={[Function]}
        hazardDetails={Array []}
        hazardReportDetails={Array []}
        incidentCategories={Array []}
        incidentReportDetails={Array []}
        invoiceAction={
          Object {
            "getPreviewInvoiceList": [Function],
          }
        }
        invoiceDetails={Array []}
        invoices={Array []}
        jobDetails={Object {}}
        jobReports={Array []}
        keepDirtyOnReinitialize={false}
        orgUserActions={
          Object {
            "deleteOrganizationUser": [Function],
            "getOrganizationUsers": [Function],
            "inviteUsers": [Function],
            "resendAtrributeCode": [Function],
            "updateOrganizationUser": [Function],
            "updateUsersFromView": [Function],
          }
        }
        persistentSubmitErrors={false}
        pure={true}
        riskControls={Array []}
        sAJobMgmtAction={
          Object {
            "addJobSWMSSign": [Function],
            "deleteJobScheduleShift": [Function],
            "deletePhotosDocsFile": [Function],
            "deleteReport": [Function],
            "getJobDetails": [Function],
            "getJobMembers": [Function],
            "getJobReport": [Function],
            "getJobSignOff": [Function],
            "getPhotosDocsFileList": [Function],
            "getSWMSSignDetails": [Function],
            "getSingleUserLicences": [Function],
            "lockSignatureProcess": [Function],
            "saveJobReport": [Function],
            "sendEmailJobReport": [Function],
            "signOffJobSheet": [Function],
            "updateJob": [Function],
            "updateJobReport": [Function],
            "updateJobScheduleShift": [Function],
            "updateUserSiteTime": [Function],
          }
        }
        shouldAsyncValidate={[Function]}
        shouldError={[Function]}
        shouldValidate={[Function]}
        shouldWarn={[Function]}
        store={
          Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
        }
        submitAsSideEffect={false}
        timeSheetAction={
          Object {
            "approveJobReport": [Function],
            "approveTimesheet": [Function],
            "deleteSingleStaff": [Function],
            "getCompletedJobList": [Function],
            "getInvoices": [Function],
            "getStaffTimeSheetList": [Function],
            "saveInvoice": [Function],
            "saveTimesheet": [Function],
            "updateSingleStaff": [Function],
          }
        }
        timeSheetList={Array []}
        touchOnBlur={true}
        touchOnChange={false}
        updateUnregisteredFields={false}
        validate={[Function]}
      />,
      "value": null,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "SAIncidentReportAction": Object {
          "AssignCorrectiveAction": [Function],
          "AssignHazard": [Function],
          "addHazardReport": [Function],
          "addIncidentReport": [Function],
          "getActionAssign": [Function],
          "getAllBeforeConsequences": [Function],
          "getAllHazards": [Function],
          "getAllHazardsByJobId": [Function],
          "getAllHazardsNew": [Function],
          "getAllIncidents": [Function],
          "getAllIncidentsByJobId": [Function],
          "getAllLikelihoodBeforeControl": [Function],
          "getHazardDetails": [Function],
          "getHazardDetailsNew": [Function],
          "getIncidentCategories": [Function],
          "getIncidentDetails": [Function],
          "getRiskControls": [Function],
        },
        "allHazardReportsByJob": Array [],
        "allIncidentReportsByJob": Array [],
        "destroyOnUnmount": true,
        "enableReinitialize": true,
        "filePath": "",
        "forceUnregisterOnUnmount": false,
        "form": "showTimeSheet",
        "formValues": Object {},
        "getFormState": [Function],
        "hazardDetails": Array [],
        "hazardReportDetails": Array [],
        "incidentCategories": Array [],
        "incidentReportDetails": Array [],
        "invoiceAction": Object {
          "getPreviewInvoiceList": [Function],
        },
        "invoiceDetails": Array [],
        "invoices": Array [],
        "jobDetails": Object {},
        "jobReports": Array [],
        "keepDirtyOnReinitialize": false,
        "orgUserActions": Object {
          "deleteOrganizationUser": [Function],
          "getOrganizationUsers": [Function],
          "inviteUsers": [Function],
          "resendAtrributeCode": [Function],
          "updateOrganizationUser": [Function],
          "updateUsersFromView": [Function],
        },
        "persistentSubmitErrors": false,
        "pure": true,
        "riskControls": Array [],
        "sAJobMgmtAction": Object {
          "addJobSWMSSign": [Function],
          "deleteJobScheduleShift": [Function],
          "deletePhotosDocsFile": [Function],
          "deleteReport": [Function],
          "getJobDetails": [Function],
          "getJobMembers": [Function],
          "getJobReport": [Function],
          "getJobSignOff": [Function],
          "getPhotosDocsFileList": [Function],
          "getSWMSSignDetails": [Function],
          "getSingleUserLicences": [Function],
          "lockSignatureProcess": [Function],
          "saveJobReport": [Function],
          "sendEmailJobReport": [Function],
          "signOffJobSheet": [Function],
          "updateJob": [Function],
          "updateJobReport": [Function],
          "updateJobScheduleShift": [Function],
          "updateUserSiteTime": [Function],
        },
        "shouldAsyncValidate": [Function],
        "shouldError": [Function],
        "shouldValidate": [Function],
        "shouldWarn": [Function],
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "submitAsSideEffect": false,
        "timeSheetAction": Object {
          "approveJobReport": [Function],
          "approveTimesheet": [Function],
          "deleteSingleStaff": [Function],
          "getCompletedJobList": [Function],
          "getInvoices": [Function],
          "getStaffTimeSheetList": [Function],
          "saveInvoice": [Function],
          "saveTimesheet": [Function],
          "updateSingleStaff": [Function],
        },
        "timeSheetList": Array [],
        "touchOnBlur": true,
        "touchOnChange": false,
        "updateUnregisteredFields": false,
        "usersList": undefined,
        "validate": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": Object {
          "destroyOnUnmount": true,
          "enableReinitialize": true,
          "forceUnregisterOnUnmount": false,
          "form": "showTimeSheet",
          "getFormState": [Function],
          "keepDirtyOnReinitialize": false,
          "persistentSubmitErrors": false,
          "pure": true,
          "shouldAsyncValidate": [Function],
          "shouldError": [Function],
          "shouldValidate": [Function],
          "shouldWarn": [Function],
          "submitAsSideEffect": false,
          "touchOnBlur": true,
          "touchOnChange": false,
          "updateUnregisteredFields": false,
          "validate": [Function],
        },
        "displayName": "ReduxForm",
        "render": [Function],
        Symbol(Symbol.hasInstance): [Function],
      },
    },
    "type": Object {
      "$$typeof": Symbol(react.provider),
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": Object {
          "$$typeof": Symbol(react.context),
          "_calculateChangedBits": null,
          "_context": [Circular],
        },
        "Provider": [Circular],
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ForwardRef
          SAIncidentReportAction={
            Object {
              "AssignCorrectiveAction": [Function],
              "AssignHazard": [Function],
              "addHazardReport": [Function],
              "addIncidentReport": [Function],
              "getActionAssign": [Function],
              "getAllBeforeConsequences": [Function],
              "getAllHazards": [Function],
              "getAllHazardsByJobId": [Function],
              "getAllHazardsNew": [Function],
              "getAllIncidents": [Function],
              "getAllIncidentsByJobId": [Function],
              "getAllLikelihoodBeforeControl": [Function],
              "getHazardDetails": [Function],
              "getHazardDetailsNew": [Function],
              "getIncidentCategories": [Function],
              "getIncidentDetails": [Function],
              "getRiskControls": [Function],
            }
          }
          allHazardReportsByJob={Array []}
          allIncidentReportsByJob={Array []}
          destroyOnUnmount={true}
          enableReinitialize={true}
          filePath=""
          forceUnregisterOnUnmount={false}
          form="showTimeSheet"
          formValues={Object {}}
          getFormState={[Function]}
          hazardDetails={Array []}
          hazardReportDetails={Array []}
          incidentCategories={Array []}
          incidentReportDetails={Array []}
          invoiceAction={
            Object {
              "getPreviewInvoiceList": [Function],
            }
          }
          invoiceDetails={Array []}
          invoices={Array []}
          jobDetails={Object {}}
          jobReports={Array []}
          keepDirtyOnReinitialize={false}
          orgUserActions={
            Object {
              "deleteOrganizationUser": [Function],
              "getOrganizationUsers": [Function],
              "inviteUsers": [Function],
              "resendAtrributeCode": [Function],
              "updateOrganizationUser": [Function],
              "updateUsersFromView": [Function],
            }
          }
          persistentSubmitErrors={false}
          pure={true}
          riskControls={Array []}
          sAJobMgmtAction={
            Object {
              "addJobSWMSSign": [Function],
              "deleteJobScheduleShift": [Function],
              "deletePhotosDocsFile": [Function],
              "deleteReport": [Function],
              "getJobDetails": [Function],
              "getJobMembers": [Function],
              "getJobReport": [Function],
              "getJobSignOff": [Function],
              "getPhotosDocsFileList": [Function],
              "getSWMSSignDetails": [Function],
              "getSingleUserLicences": [Function],
              "lockSignatureProcess": [Function],
              "saveJobReport": [Function],
              "sendEmailJobReport": [Function],
              "signOffJobSheet": [Function],
              "updateJob": [Function],
              "updateJobReport": [Function],
              "updateJobScheduleShift": [Function],
              "updateUserSiteTime": [Function],
            }
          }
          shouldAsyncValidate={[Function]}
          shouldError={[Function]}
          shouldValidate={[Function]}
          shouldWarn={[Function]}
          store={
            Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            }
          }
          submitAsSideEffect={false}
          timeSheetAction={
            Object {
              "approveJobReport": [Function],
              "approveTimesheet": [Function],
              "deleteSingleStaff": [Function],
              "getCompletedJobList": [Function],
              "getInvoices": [Function],
              "getStaffTimeSheetList": [Function],
              "saveInvoice": [Function],
              "saveTimesheet": [Function],
              "updateSingleStaff": [Function],
            }
          }
          timeSheetList={Array []}
          touchOnBlur={true}
          touchOnChange={false}
          updateUnregisteredFields={false}
          validate={[Function]}
        />,
        "value": null,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "SAIncidentReportAction": Object {
            "AssignCorrectiveAction": [Function],
            "AssignHazard": [Function],
            "addHazardReport": [Function],
            "addIncidentReport": [Function],
            "getActionAssign": [Function],
            "getAllBeforeConsequences": [Function],
            "getAllHazards": [Function],
            "getAllHazardsByJobId": [Function],
            "getAllHazardsNew": [Function],
            "getAllIncidents": [Function],
            "getAllIncidentsByJobId": [Function],
            "getAllLikelihoodBeforeControl": [Function],
            "getHazardDetails": [Function],
            "getHazardDetailsNew": [Function],
            "getIncidentCategories": [Function],
            "getIncidentDetails": [Function],
            "getRiskControls": [Function],
          },
          "allHazardReportsByJob": Array [],
          "allIncidentReportsByJob": Array [],
          "destroyOnUnmount": true,
          "enableReinitialize": true,
          "filePath": "",
          "forceUnregisterOnUnmount": false,
          "form": "showTimeSheet",
          "formValues": Object {},
          "getFormState": [Function],
          "hazardDetails": Array [],
          "hazardReportDetails": Array [],
          "incidentCategories": Array [],
          "incidentReportDetails": Array [],
          "invoiceAction": Object {
            "getPreviewInvoiceList": [Function],
          },
          "invoiceDetails": Array [],
          "invoices": Array [],
          "jobDetails": Object {},
          "jobReports": Array [],
          "keepDirtyOnReinitialize": false,
          "orgUserActions": Object {
            "deleteOrganizationUser": [Function],
            "getOrganizationUsers": [Function],
            "inviteUsers": [Function],
            "resendAtrributeCode": [Function],
            "updateOrganizationUser": [Function],
            "updateUsersFromView": [Function],
          },
          "persistentSubmitErrors": false,
          "pure": true,
          "riskControls": Array [],
          "sAJobMgmtAction": Object {
            "addJobSWMSSign": [Function],
            "deleteJobScheduleShift": [Function],
            "deletePhotosDocsFile": [Function],
            "deleteReport": [Function],
            "getJobDetails": [Function],
            "getJobMembers": [Function],
            "getJobReport": [Function],
            "getJobSignOff": [Function],
            "getPhotosDocsFileList": [Function],
            "getSWMSSignDetails": [Function],
            "getSingleUserLicences": [Function],
            "lockSignatureProcess": [Function],
            "saveJobReport": [Function],
            "sendEmailJobReport": [Function],
            "signOffJobSheet": [Function],
            "updateJob": [Function],
            "updateJobReport": [Function],
            "updateJobScheduleShift": [Function],
            "updateUserSiteTime": [Function],
          },
          "shouldAsyncValidate": [Function],
          "shouldError": [Function],
          "shouldValidate": [Function],
          "shouldWarn": [Function],
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "submitAsSideEffect": false,
          "timeSheetAction": Object {
            "approveJobReport": [Function],
            "approveTimesheet": [Function],
            "deleteSingleStaff": [Function],
            "getCompletedJobList": [Function],
            "getInvoices": [Function],
            "getStaffTimeSheetList": [Function],
            "saveInvoice": [Function],
            "saveTimesheet": [Function],
            "updateSingleStaff": [Function],
          },
          "timeSheetList": Array [],
          "touchOnBlur": true,
          "touchOnChange": false,
          "updateUnregisteredFields": false,
          "usersList": undefined,
          "validate": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "destroyOnUnmount": true,
            "enableReinitialize": true,
            "forceUnregisterOnUnmount": false,
            "form": "showTimeSheet",
            "getFormState": [Function],
            "keepDirtyOnReinitialize": false,
            "persistentSubmitErrors": false,
            "pure": true,
            "shouldAsyncValidate": [Function],
            "shouldError": [Function],
            "shouldValidate": [Function],
            "shouldWarn": [Function],
            "submitAsSideEffect": false,
            "touchOnBlur": true,
            "touchOnChange": false,
            "updateUnregisteredFields": false,
            "validate": [Function],
          },
          "displayName": "ReduxForm",
          "render": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.provider),
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": Object {
            "$$typeof": Symbol(react.context),
            "_calculateChangedBits": null,
            "_context": [Circular],
          },
          "Provider": [Circular],
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
